starknet_devnet/server.py

```diff
@@ -319,14 +319,15 @@ def main():
         except (FileNotFoundError, pickle.UnpicklingError):
             sys.exit(f"Error: Cannot load from {args.load_path}. Make sure the file exists and contains a Devnet dump.")
 
-    if args.dump_on == DumpOn.EXIT:
-        for sig in [signal.SIGTERM, signal.SIGINT]:
-            signal.signal(sig, dump_on_exit)
-
     dumper.dump_path = args.dump_path
     dumper.dump_on = args.dump_on
-
-    app.run(host=args.host, port=args.port)
+    starknet_wrapper.lite_mode = args.lite_mode
+    try:
+        meinheld.listen((args.host, args.port))
+        meinheld.run(app)
+    finally:
+        if args.dump_on == DumpOn.EXIT:
+            dump_on_exit()
```

Comment:
If `dump_on_exit` isn't used anywhere else, you can remove it and replace this call with `dumper.dump(dumper.dump_path)`.